pkg/controller/servicebindingrequest/service.go

```diff
@@ -0,0 +1,220 @@
+package servicebindingrequest
+
+import (
+	"errors"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/annotations"
+	"github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/nested"
+)
+
+var (
+	// ErrUnspecifiedBackingServiceNamespace is returned when the namespace of a service is
+	// unspecified.
+	ErrUnspecifiedBackingServiceNamespace = errors.New("backing service namespace is unspecified")
+	// EmptyBackingServiceSelectorsErr is returned when no backing service selectors have been
+	// informed in the Service Binding Request.
+	ErrEmptyBackingServiceSelectors = errors.New("backing service selectors are empty")
+)
+
+func findService(
+	client dynamic.Interface,
+	ns string,
+	gvk schema.GroupVersionKind,
+	resourceRef string,
+) (
+	*unstructured.Unstructured,
+	error,
+) {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+
+	if len(ns) == 0 {
+		return nil, ErrUnspecifiedBackingServiceNamespace
+	}
+
+	// delegate the search selector's namespaced resource client
+	return client.
+		Resource(gvr).
+		Namespace(ns).
+		Get(resourceRef, metav1.GetOptions{})
+}
+
+// CRDGVR is the plural GVR for Kubernetes CRDs.
+var CRDGVR = schema.GroupVersionResource{
+	Group:    "apiextensions.k8s.io",
+	Version:  "v1beta1",
+	Resource: "customresourcedefinitions",
+}
+
+func findServiceCRD(client dynamic.Interface, gvk schema.GroupVersionKind) (*unstructured.Unstructured, error) {
+	// gvr is the plural guessed resource for the given GVK
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	// crdName is the string'fied GroupResource, e.g. "deployments.apps"
+	crdName := gvr.GroupResource().String()
+	// delegate the search to the CustomResourceDefinition resource client
+	return client.Resource(CRDGVR).Get(crdName, metav1.GetOptions{})
+}
+
+func loadDescriptor(anns map[string]string, path string, descriptor string, root string) {
```

Comment:
since you use `anns` here for annotations, you could use the same name here as well: https://github.com/redhat-developer/service-binding-operator/blob/ac35badbc6321c1f0b2f3b5dd116361ca65e81d8/pkg/controller/servicebindingrequest/olm.go#L238